src/flask_session/__init__.py

```diff
@@ -45,53 +55,144 @@ def __init__(self, app=None):
     def init_app(self, app):
         """This is used to set up session for your app object.
 
+        When app.session_interface is None
+        or
+        app.config['SESSION_TYPE'] == 'custom'
+        we using custom implementation from application.
+
         :param app: the Flask app object with proper configuration.
         """
-        app.session_interface = self._get_interface(app)
+        if app.session_interface is None or (
+            "SESSION_TYPE" in app.config and app.config["SESSION_TYPE"] != "custom"
+        ):
+            app.session_interface = self._get_interface(app)
 
     def _get_interface(self, app):
         config = app.config.copy()
-        config.setdefault('SESSION_TYPE', 'null')
-        config.setdefault('SESSION_PERMANENT', True)
-        config.setdefault('SESSION_USE_SIGNER', False)
-        config.setdefault('SESSION_KEY_PREFIX', 'session:')
-        config.setdefault('SESSION_REDIS', None)
-        config.setdefault('SESSION_MEMCACHED', None)
-        config.setdefault('SESSION_FILE_DIR',
-                          os.path.join(os.getcwd(), 'flask_session'))
-        config.setdefault('SESSION_FILE_THRESHOLD', 500)
-        config.setdefault('SESSION_FILE_MODE', 384)
-        config.setdefault('SESSION_MONGODB', None)
-        config.setdefault('SESSION_MONGODB_DB', 'flask_session')
-        config.setdefault('SESSION_MONGODB_COLLECT', 'sessions')
-        config.setdefault('SESSION_SQLALCHEMY', None)
-        config.setdefault('SESSION_SQLALCHEMY_TABLE', 'sessions')
-
-        if config['SESSION_TYPE'] == 'redis':
+        config.setdefault("SESSION_TYPE", "null")
```

Comment:
Thanks for making the first cut, @christopherpickering ! This PR used to have more than 10K lines, now it reduces to 1.5K.

But it is still somewhat large. From a tactical standpoint, I would suggest to create another PR with styling changes only, which would be much straightforward to review and merge. And then we can rebase this PR which will then be much smaller. You do not have to create that style PR, I can do that for you/us, if you prefer. Just let me know. By the way, what linter do you use?

Reply:
I don't have time to spend on the project, I don't even use flask sessions anymore :)

I normally use black with pre-commit. I think this code hasn't been passed through a formatter yet?

Thanks!